python/cudf/cudf/core/dataframe.py

```diff
@@ -3375,37 +3308,37 @@ def from_pandas(cls, dataframe, nan_as_null=True):
 
         df = cls()
         # Set columns
-        for colk in dataframe.columns:
+        for i, colk in enumerate(dataframe.columns):
             vals = dataframe[colk].values
             # necessary because multi-index can return multiple
             # columns for a single key
             if len(vals.shape) == 1:
-                df[colk] = Series(vals, nan_as_null=nan_as_null)
+                df[i] = Series(vals, nan_as_null=nan_as_null)
             else:
                 vals = vals.T
                 if vals.shape[0] == 1:
-                    df[colk] = Series(vals.flatten(), nan_as_null=nan_as_null)
+                    df[i] = Series(vals.flatten(), nan_as_null=nan_as_null)
                 else:
                     # TODO fix multiple column with same name with different
                     # method.
```

Comment:
Do we need to address this / throw some type of exception now?

Reply:
Just removed this comment as I'm not sure what the downstream implications are. We can fix if it comes up